cmd/api-server/main.go

```diff
@@ -313,6 +316,74 @@ func main() {
 	proContext, err := commons.ReadProContext(ctx, cfg, grpcClient)
 	commons.ExitOnError("cannot connect to control plane", err)
 
+	// SUPER AGENT DEPRECATION MIGRATION
+	// "Super" Agents are deprecated, instead they are being migrated to a more generic Agent with Capabilities.
+	// The migration can only occur if:
+	// - The Control Plane supports being a Source of Truth.
+	// - The current Agent is still considered to be a Super Agent by the Control Plane.
+	// - The current Agent is not being held back as a Super Agent by an override.
+	// Once Super Agent migration has been completed across all clients then this entire block can be removed.
+	if proContext.CloudStorageSupportedInControlPlane && proContext.Agent.IsSuperAgent && !cfg.ForceSuperAgentMode {
+		// Run migration in a separate goroutine so that it does not affect other operations.
```

Comment:
Isn't the whole point of One-Shot migration that we do it blocking and then it's done with. What I mean with this is that we could start looking in removing superagent specific code instead of needing to keep this backwards compatible for 3-6 months.